plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -271,42 +289,66 @@ public <T> T getAdapter(Class<T> adapter) {
     {
-        try {
-            // Load contents in two steps (empty + real in async mode). Workaround for some strange bug in StyledText in E4.13 (#6701)
-            final TextViewer textViewer = editor.getTextViewer();
-            final String encoding = getPanelSettings().get(PREF_TEXT_EDITOR_ENCODING);
-            final ContentEditorInput textInput = new ContentEditorInput(valueController, null, null, encoding, monitor);
-            boolean longContent = textInput.getContentLength() > LONG_CONTENT_LENGTH;
-            if (longContent) {
-                UIUtils.asyncExec(() -> {
-                    editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
-                });
-            }
-            UIUtils.asyncExec(() -> {
-
-                if (textViewer != null && editor != null) {
-                    StyledText textWidget = textViewer.getTextWidget();
-                    if (textWidget != null && longContent) {
-                        GC gc = new GC(textWidget);
-                        try {
-                            UIUtils.drawMessageOverControl(textWidget, gc, NLS.bind(ResultSetMessages.panel_editor_text_loading_placeholder_label, textInput.getContentLength()), 0);
-                            editor.setInput(textInput);
-                        } finally {
-                            gc.dispose();
-                        }
-                    } else {
-                        editor.setInput(textInput);
-                    }
-                    applyEditorStyle();
-                }
-            });
-        } catch (Exception e) {
-            throw new DBException("Error loading text value", e);
-        } finally {
-            monitor.done();
-        }
+    	try {
+			int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES)*1000;
+			if (editor == null) {
+				log.error("Editor is null or undefined");
+				return;
+			}
+			resetEditorInput();
+			byte[] bytes = CommonUtils.toString(value).getBytes();
+			if (bytes.length > maxContentSize) {
+				showRestrictedContent(editorPaintListener, bytes);
+			} else {
+				showRegularContent(editorPaintListener, bytes, monitor);
+			}
+
+		} catch (Exception e) {
+			throw new DBException("Error loading text value", e);
+		} finally {
+			monitor.done();
+		}
     }
 
+    private void resetEditorInput() {
+		// Load contents in two steps (empty + real in async mode). Workaround for some
+		// strange bug in StyledText in E4.13 (#6701)
+		UIUtils.asyncExec(() -> {
+			if (editor != null) {
+				editorPaintListener.updateStatusMessage("");
+				editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
+			}
+		});
+	}
+
+	private void showRegularContent(final PanelEditorPaintListener editorPaintListener, byte[] bytes,
+			DBRProgressMonitor monitor) throws DBException {
```

Comment:
We use such formatting for a long lines
```suggestion
	private void showRegularContent(
	    @NotNull final PanelEditorPaintListener editorPaintListener,
	    byte[] bytes,
	    @NotNull DBRProgressMonitor monitor
	) throws DBException {
```

Reply:
Let's have agrement of code style, Im currently  using default eclipse settings.

Reply:
We have it already.

Such formatting for long lines
Long line is 140 symbols and more
Formatting for casts is the following `(ClassName) varName` - space in between
Arrays formatting: `new String[]{"*.*", "*.jks", "*"}`
Don't use static imports
Also note that we use nullable annotations.

Reply:
Also we don't write if or cycles without blocks. For example:
```
if (true) System.out.println("Hey!");
```
should be 
```
if (true) {
    System.out.println("Hey!");
}
```

Reply:
It is clear, same for each IDE,
one more point, the max line width is 140  instead of 120 (120 by defaul) is it correct?